File: backend/app/models/email.py
"""Email model."""

from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text, Float
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Email(Base):
    """Represents a synced Gmail message."""

    __tablename__ = "emails"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), index=True)
    gmail_id: Mapped[str] = mapped_column(String(255), unique=True, index=True)
    subject: Mapped[str | None] = mapped_column(String(500), nullable=True)
    sender: Mapped[str | None] = mapped_column(String(255), nullable=True)
    snippet: Mapped[str | None] = mapped_column(Text, nullable=True)
    body: Mapped[str | None] = mapped_column(Text, nullable=True)
    body_text: Mapped[str | None] = mapped_column(Text, nullable=True)
    body_html: Mapped[str | None] = mapped_column(Text, nullable=True)
    body_fetched_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    has_attachments: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    received_at: Mapped[datetime] = mapped_column(DateTime, nullable=False)
    category: Mapped[str | None] = mapped_column(String(50), nullable=True)
    ai_summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    ai_confidence: Mapped[float | None] = mapped_column(Float, nullable=True)
    is_unread: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    is_archived: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    synced_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

    user = relationship("User", back_populates="emails")
    ai_queue = relationship("AIQueue", back_populates="email", uselist=False, cascade="all, delete-orphan")

File: ios/InboxIQ/InboxIQ.backup-before-integration/Models/Email.swift
import Foundation

struct Email: Identifiable, Hashable, Codable {
    let id: UUID
    let gmailId: String
    let subject: String
    let sender: String
    let snippet: String
    let category: Category?
    let aiSummary: String?
    let aiConfidence: Double?
    let receivedAt: Date
    let syncedAt: Date
    var isUnread: Bool

    init(
        id: UUID = UUID(),
        gmailId: String,
        subject: String,
        sender: String,
        snippet: String,
        category: Category? = nil,
        aiSummary: String? = nil,
        aiConfidence: Double? = nil,
        receivedAt: Date,
        syncedAt: Date,
        isUnread: Bool = true
    ) {
        self.id = id
        self.gmailId = gmailId
        self.subject = subject
        self.sender = sender
        self.snippet = snippet
        self.category = category
        self.aiSummary = aiSummary
        self.aiConfidence = aiConfidence
        self.receivedAt = receivedAt
        self.syncedAt = syncedAt
        self.isUnread = isUnread
    }
}
